src/shared/components/common/subscribe-button.tsx

```diff
@@ -0,0 +1,208 @@
+import { validInstanceTLD } from "@utils/helpers";
+import classNames from "classnames";
+import { NoOptionI18nKeys } from "i18next";
+import { Component, MouseEventHandler, linkEvent } from "inferno";
+import { CommunityView } from "lemmy-js-client";
+import { I18NextService, UserService } from "../../services";
+import { VERSION } from "../../version";
+import { Icon, Spinner } from "./icon";
+
+interface SubscribeButtonProps {
+  communityView: CommunityView;
+  onFollow: MouseEventHandler;
+  onUnFollow: MouseEventHandler;
+  loading: boolean;
+}
+
+export function SubscribeButton({
+  communityView: {
+    subscribed,
+    community: { actor_id },
+  },
+  onFollow,
+  onUnFollow,
+  loading,
+}: SubscribeButtonProps) {
+  let i18key: NoOptionI18nKeys;
+
+  switch (subscribed) {
+    case "NotSubscribed": {
+      i18key = "subscribe";
+
+      break;
+    }
+    case "Subscribed": {
+      i18key = "joined";
+
+      break;
+    }
+    default: {
+      i18key = "subscribe_pending";
+
+      break;
+    }
+  }
+
+  const buttonClass = "btn d-block mb-2 w-100";
+
+  if (!UserService.Instance.myUserInfo) {
+    return (
+      <>
+        <button
+          type="button"
+          className={classNames(buttonClass, "btn-secondary")}
+          data-bs-toggle="modal"
+          data-bs-target="#remoteFetchModal"
+        >
+          {I18NextService.i18n.t("subscribe")}
+        </button>
+        <RemoteFetchModal communityActorId={actor_id} />
+      </>
+    );
+  }
+
+  return (
+    <button
+      type="button"
+      className={classNames(
+        buttonClass,
+        `btn-${subscribed === "Pending" ? "warning" : "secondary"}`,
+      )}
+      onClick={subscribed === "NotSubscribed" ? onFollow : onUnFollow}
+    >
+      {loading ? (
+        <Spinner />
+      ) : (
+        <>
+          {subscribed === "Subscribed" && (
+            <Icon icon="check" classes="icon-inline me-1" />
+          )}
+          {I18NextService.i18n.t(i18key)}
+        </>
+      )}
+    </button>
+  );
+}
+
+interface RemoteFetchModalProps {
+  communityActorId: string;
+}
+
+interface RemoteFetchModalState {
+  instanceText: string;
+}
+
+function handleInput(i: RemoteFetchModal, event: any) {
+  i.setState({ instanceText: event.target.value });
+}
+
+function focusInput() {
+  document.getElementById("remoteFetchInstance")?.focus();
+}
+
+function submitRemoteFollow(
+  { state: { instanceText }, props: { communityActorId } }: RemoteFetchModal,
+  event: Event,
+) {
+  event.preventDefault();
+
+  if (!validInstanceTLD(instanceText)) {
+    // TODO: Figure out appropriate way of handling invalid domainss
+    console.log("I should do something about this.");
+  }
```

Comment:
This will need to be handled. @dessalines will a toast message about the error suffice?

Reply:
Yep: `toast(ERR_STR, "danger");`